adaptivefiltering/dataset.py

```diff
@@ -38,6 +41,80 @@ def __init__(self, filename, warning_threshold=750000):
                 )
             )
 
+    def generate_geoTif(
+        self,
+        filename,
+        resolution=2.0,
+    ):
+        """Calculate and safe a meshgrid of the dataset with a custom resolution.
+
+        :param filename:
+            Filename to save the .tif. This can be a relative or absolute path
+        :type filename: str
+        :param resolution:
+            The resolution used to calculate the mesh from the point cloud.
+        :type resolution: float
+        """
+        # if .tif is already in the filename it will be removed to avoid double file extension
+        if filename.endswith(".tif"):
+            filename = filename.split(".tif")[0]
+        # configure the geotif pipeline
+        geotif_pipeline_json = [
+            self.filename,
+            {
+                "filename": filename + ".tif",
+                "gdaldriver": "GTiff",
+                "output_type": "all",
+                "resolution": resolution,
+                "type": "writers.gdal",
+            },
+        ]
+        # setup and execute the geotif pipeline
+        pipeline = pdal.Pipeline(json.dumps(geotif_pipeline_json))
+        geotif_pipeline = pipeline.execute()
+
+    def show_mesh(self, filename=None, resolution=2.0):
+        """Load an existing . tif file or create a temporary file with a given resolution. The -tif is than visualised as a 3d mesh
+
+        :param filename:
+            optional filename to load a specific .tif file. The filename uses the same search method as the base class.
+            A .tif ending must be used.
```

Comment:
I thought a while about this parameter. Most of the time, users that call `show_mesh` really only want to see the mesh (e.g. in an optimization loop) and are not really interested in where the file is written. *IF* they are interested, there is the `save_mesh` method exactly for that. I would therefore prefer the following approach:
* `show_mesh` has no `filename` parameter
* If `save_mesh` was called before `show_mesh`, the saved file is loaded instead of recreating as a temporary.

Reply:
My thinking was, that when we calculate different filter settings we get multiple files that we want to show at the same time without having to again load each .laz file into memory again. 
What one could do is split this into 4 functions:
write_mesh/tif: creates a geotiff file in memory 
read_mesh/tif: read a geotif ffile to memory
save_mesh/tif: saves the stored geotiff to the drive, this would call write_mesh/tif.
show_mesh/tif: same as before, but only using the .tif that is currently in memory. 

This way one could choose if one wants to save a file or just look at it, but one can also quickly load many different files and plot them side by side.